compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3047,31 +3048,63 @@ TR::Register *OMR::X86::TreeEvaluator::directCallEvaluator(TR::Node *node, TR::C
       {
       TR::InstOpCode::Mnemonic op = TR::InstOpCode::bad;
 
-      if (comp->getSymRefTab()->isNonHelper(SymRef, TR::SymbolReferenceTable::atomicAddSymbol))
+      if (symRefTab->isNonHelper(SymRef, TR::SymbolReferenceTable::atomicAddSymbol))
          {
          op = node->getChild(1)->getDataType().isInt32() ? TR::InstOpCode::LADD4MemReg : TR::InstOpCode::LADD8MemReg;
          }
-      else if (comp->getSymRefTab()->isNonHelper(SymRef, TR::SymbolReferenceTable::atomicFetchAndAddSymbol))
+      else if (symRefTab->isNonHelper(SymRef, TR::SymbolReferenceTable::atomicFetchAndAddSymbol))
          {
-         op = node->getChild(1)->getDataType().isInt32() ? TR::InstOpCode::LXADD4MemReg : TR::InstOpCode::LXADD8MemReg;
+         auto dataType = node->getChild(1)->getDataType();
+         if (dataType.isInt8())
+            op = TR::InstOpCode::LXADD1MemReg;
+         else if (dataType.isInt16())
+            op = TR::InstOpCode::LXADD2MemReg;
+         else if (dataType.isInt32())
+            op = TR::InstOpCode::LXADD4MemReg;
+         else
+            op = TR::InstOpCode::LXADD8MemReg;
+         }
+      else if (symRefTab->isNonHelper(SymRef, TR::SymbolReferenceTable::atomicFetchAndAdd8BitSymbol))
```

Comment:
To allow downstream projects to check whether the `atomicFetchAndAdd8BitSymbol` and `atomicFetchAndAdd16BitSymbol` non-helpers are supported on x86, you'll need to update the implementation of [`OMR::X86::CodeGenerator::supportsNonHelper`](https://github.com/eclipse/omr/blob/daa7c85bae82a3f72af5053b1dc51f5f6217f20d/compiler/x/codegen/OMRCodeGenerator.cpp#L1269-L1283).  Similarly for Power you'll need to update [`OMR::Power::CodeGenerator::supportsNonHelper`](https://github.com/eclipse/omr/blob/daa7c85bae82a3f72af5053b1dc51f5f6217f20d/compiler/p/codegen/OMRCodeGenerator.cpp#L2436-L2456).